Editor/Unity/PreBuildDotNetChecker.cs

```diff
@@ -0,0 +1,27 @@
+﻿using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+
+namespace Innoactive.CreatorEditor
+{
+    /// <summary>
+    /// Pre-process validation to identify if the 'API Compatibility Level' is set to '.Net 4.x'.
+    /// </summary>
+    internal class PreBuildDotNetChecker : IPreprocessBuildWithReport
+    {
+        ///<inheritdoc />
+        public int callbackOrder => 0;
+
+        ///<inheritdoc />
+        public void OnPreprocessBuild(BuildReport report)
+        {
+            ApiCompatibilityLevel currentLevel = EditorUtils.GetCurrentCompatibilityLevel();
+
+            if (currentLevel != ApiCompatibilityLevel.NET_4_6)
+            {
+                DotNetWizard.OpenWizard();
+                throw new BuildPlayerWindow.BuildMethodException("This Unity project is not configured for .Net 4.X, but the Creator SDK requires .NET 4.X support.");
```

Comment:
Do not like the idea of throwing an error. AFAIK the build fails because of the serializer, right? So if somebody uses his own serializer, which we specifically implemented, then he cannot build with 2.0 even though it would not do anything. 
So a warning might be enough? Or taking this thing out in general because we already notify the user before

Reply:
Throwing an error is the only way to prevent the build.
A warning could be easily ignored since the console shows tons of them after compiling.

Reply:
Then I would leave the pre-build check completely. Restricting the user is no fun

Reply:
Let me think if we can implement this feature in another way 🤔